src/Navigation/Dropdown/Dropdown.tsx

```diff
@@ -17,231 +17,180 @@ import {
 
 import DropdownItem, { Props as DropdownItemProps } from './DropdownItem';
 
-class Dropdown extends React.Component<Props, State> {
-  static Item = DropdownItem;
-
-  static defaultProps = {
-    showHoverLine: false,
-    dropDownPlacement: 'left',
-    noLineBreak: false,
-    iconDefaultColor: 'black',
-    showFullWidth: false,
-  };
-
-  dropdownBodyRef: React.RefObject<HTMLUListElement>;
-  handleEscKeydown: (this: Document, ev: KeyboardEvent) => any;
-
-  constructor(props: Props) {
-    super(props);
-    this.dropdownBodyRef = React.createRef();
-    this.handleEscKeydown = escEvent(this.handleClose);
-    this.state = {
-      dropdownLabel: props.label,
-      isOpen: false,
-      cursor: 0,
-    };
-  }
-
-  handleOpen = () => {
-    const { disabled } = this.props;
-    const { isOpen } = this.state;
-
+const Dropdown = (props: Props) => {
+  const {
+    label,
+    showHoverLine = false,
+    dropDownPlacement = 'left',
+    noLineBreak = false,
+    iconDefaultColor = 'black',
+    showFullWidth = false,
+    disabled,
+    hoverToOpen,
+    onChange,
+    children,
+    leftIcon,
+    className,
+    itemElement,
+    ...defaultProps
+  } = props;
+
+  const dropdownBodyRef = React.useRef<HTMLUListElement>(null);
+
+  const [dropdownLabel, setDropdownLabel] = React.useState<string>(label);
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const [cursor, setCursor] = React.useState<number>(0);
+
+  const handleOpen = () => {
     if (!disabled) {
-      this.setState({
-        isOpen: !isOpen,
-      });
+      setIsOpen(!isOpen);
     }
   };
 
-  hoverOpen = () => {
-    const { hoverToOpen } = this.props;
-
+  const hoverOpen = () => {
     if (hoverToOpen) {
-      this.setState({
-        isOpen: true,
-      });
+      setIsOpen(true);
     }
   };
 
-  hoverClose = () => {
-    const { hoverToOpen } = this.props;
-
+  const hoverClose = () => {
     if (hoverToOpen) {
-      this.setState({
-        isOpen: false,
-        cursor: 0,
-      });
+      setIsOpen(false);
+      setCursor(0);
     }
   };
 
-  handleClose = () => {
-    this.setState({
-      isOpen: false,
-      cursor: 0,
-    });
+  const handleClose = () => {
+    setIsOpen(false);
+    setCursor(0);
   };
 
-  handleClickItem = (
+  const handleClickItem = (
     onClick: (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => void
   ) => (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-    const { onChange } = this.props;
     const itemElement = e.target as HTMLLIElement;
 
     if (itemElement.dataset.value) {
-      this.setState({
-        dropdownLabel: itemElement.dataset.value,
-        isOpen: false,
-      });
+      // TODO: This should probably set to itemElement.innerHTML
+      setDropdownLabel(itemElement.dataset.value);
```

Comment:
As you can see, when an item is clicked, the label gets set to the item's value. Of course that makes little sense and indeed further down, where enter key presses are handled, the handler there sets the label to the item's `innerHTML`. So I'm pretty sure this is unintended.

But since this is just a conversion PR I won't change the behaviour and keep setting the label to the item's value here for now.

Reply:
do you mind open an issue for this? I think it's ok to not change it in this PR, but this is so weird haha

Reply:
Opened it here: https://github.com/glints-dev/glints-aries/issues/348